### PR TITLE
ci: 修复smoke-testing重复加载外服和国服资源导致潜在的非预期TaskData字段问题

### PR DESCRIPTION
**Pull request by status102**
修复类似 [0291163](https://github.com/MaaAssistantArknights/MaaAssistantArknights/commit/02911638ce49730d3745fd910c250bc31a9cd44c) 的问题

（本地DEBUG就先扔着了

### CODE REVIEW

**Comment by Constrat** [comment]
@status102 Is it normal that Sample.exe in Release is 212KB while in Debug it's 72KB?
Also, the release one shows all the outputs correctly (as expected). The debug only opens a terminal and then closes quickly.

**Review by Constrat — APPROVED** [review]
`DEBUG`:
- [x] YostarEN 
- [x] YostarJP
- [x] YostarKR 
- [x] txwy 
I'm guessing there is "Official"

**Review by Constrat — CHANGES_REQUESTED** [review]
I want to try and make some changes.
Wait before merge :)

**Comment by status102** [comment]
wow, but it will look like `Sample.exe` (with no param)

**Comment by Constrat** [comment]
> wow, but it will look like `Sample.exe` (with no param)

Yeah just need to add a check: if there is a param, don't run the official loader

**Comment by status102** [comment]
> Yeah just need to add a check: if there is a param, don't run the official loader

no, it is no sense. You need to load official resource FIRST. Overseas resource is incomplete.

**Comment by Constrat** [comment]
> > Yeah just need to add a check: if there is a param, don't run the official loader
> 
> no, it is no sense. You need to load official resource FIRST. Overseas resource is incomplete.

So 4 official needs to be loaded 4 times? For every smoketest?

**Comment by status102** [comment]
> So `official` needs to be loaded 4 times? For every smoketest?

yes

**Review by Constrat — APPROVED** [review]

**Comment by status102** [comment]
> So `official` needs to be loaded 4 times? For every smoketest?

Overseas resource files show resource differences between official and overseas. So we have to loaded more than once.

File: src/Cpp/main.cpp [hunk 1]
```diff
@@ -16,11 +16,22 @@ int main([[maybe_unused]] int argc, char** argv)
     // 这里默认读取的是可执行文件同目录下 resource 文件夹里的资源
     bool loaded = AsstLoadResource(cur_path.string().c_str());
     if (!loaded) {
-        std::cerr << "-------- load resource failed --------" << std::endl;
+        std::cerr << "-------- load resource failed: official --------" << std::endl;
         return -1;
     }
 
-#ifdef ASST_DEBUG
+#ifdef SMOKE_TESTING
```
Comment (zzyyyl) [inline src/Cpp/main.cpp:23]: 这玩意在哪定义了吗

Comment (status102) [reply src/Cpp/main.cpp:23]: 没找到（不知道是怎么算的，但是能跑，很怪

Comment (status102) [reply src/Cpp/main.cpp:23]: https://github.com/MaaAssistantArknights/MaaAssistantArknights/blob/dev/src%2FCpp%2FMaaSample.vcxproj#L208-L220

**Comment by status102** [comment]
没问题就合了 @zzyyyl

**Pull request merged**